File: backend/src/main/kotlin/de/vinz/openfls/domains/assistancePlans/dtos/AssistancePlanAnalysisMonthDto.kt
package de.vinz.openfls.domains.assistancePlans.dtos

import de.vinz.openfls.domains.assistancePlans.AssistancePlanHourMode
import java.time.LocalDate

data class AssistancePlanAnalysisMonthDto(
        val assistancePlanId: Long,
        val start: LocalDate,
        val end: LocalDate,
        val clientFirstName: String,
        val clientLastName: String,
        val hourMode: AssistancePlanHourMode,
        val year: Int,
        val month: Int,
        val approvedHours: Double,
        val executedHours: Double,
        val executedPercent: Double,
        val missingHours: Double)

File: frontend/src/app/pages/service-evaluation-overview/dtos/assistance-plan-analysis-month-dto.ts
import {AssistancePlanHourMode} from '../../../shared/dtos/assistance-plan-hour-mode.model';

export class AssistancePlanAnalysisMonthDto {
  assistancePlanId: number = 0
  start: string = ""
  end: string = ""
  clientFirstName: string = ""
  clientLastName: string = ""
  hourMode: AssistancePlanHourMode = AssistancePlanHourMode.EXACT
  year: number = 0
  month: number = 0
  approvedHours: number = 0
  executedHours: number = 0
  executedPercent: number = 0
  missingHours: number = 0
}
